### PR TITLE
docs: explain how to specify which npm version Renovate bot uses

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- Explain how to tell Renovate which version of npm it should use

## Context:

A user wanted to know what version of npm renovate uses for its PRs.
Turns out that there is no way to tell Renovate: "use the default version of npm that comes with the installed node version".

But @rarkins gave a solution: edit your `package.json` file to have a `engines: npm` field: https://github.com/renovatebot/renovate/discussions/8553#discussioncomment-340928

I think this solution should be documented.

This can probably be removed once Renovate bot gets a bit smarter with the npm version selection???

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/node.md [hunk 1]
```diff
@@ -55,3 +55,22 @@ We recommend that you define this support policy inside the `node` configuration
 This way, it is applied to all Node.js-related files.
 
 For additional language support see the [`supportPolicy` documentation](/configuration-options/#supportpolicy).
+
+## Configuring which version of npm Renovate uses
+
+There is no configuration option to tell Renovate to use the npm version that comes pre-installed with Node.
+
```
Comment (rarkins) [inline docs/usage/node.md:62]:
```suggestion
When `binarySource=docker`, such as in the hosted WhiteSource Renovate App, Renovate will choose and install an `npm` version dynamically.

```

File: docs/usage/node.md [hunk 2]
```diff
@@ -55,3 +55,22 @@ We recommend that you define this support policy inside the `node` configuration
...
+You must use the `engines.npm` property in your `package.json` file to specify which major version of npm your project supports.
```
Comment (rarkins) [inline docs/usage/node.md:63]:
```suggestion
To control which version or constraint is installed, you should use the `engines.npm` property in your `package.json` file.
```

File: docs/usage/node.md [hunk 3]
```diff
@@ -55,3 +55,22 @@ We recommend that you define this support policy inside the `node` configuration
...
+Renovate bot will then use that version of npm when it creates a pull request.
```
Comment (rarkins) [inline docs/usage/node.md:64]:
```suggestion
Renovate bot will then use that version constraint for npm when it creates a pull request.
```

File: docs/usage/node.md [hunk 4]
```diff
@@ -55,3 +55,22 @@ We recommend that you define this support policy inside the `node` configuration
...
+
+In your `package.json` file create a `engines` key-value pair and specify which major version of npm your project works with.
+
```
Comment (rarkins) [inline docs/usage/node.md:67]:
```suggestion
```

File: docs/usage/node.md [hunk 5]
```diff
@@ -55,3 +55,22 @@ We recommend that you define this support policy inside the `node` configuration
...
+For example, if you want to use at least npm `6.14.11` and also allow newer versions of npm in the `6.x` range, you would put this in your `package.json` file:
+
+```json
+{
+  "engines": {
+    "npm": "^6.14.11"
+  }
+}
+```
```
Comment (rarkins) [inline docs/usage/node.md:76]:
```suggestion
```

There is not currently any way to tell Renovate "just use whichever npm version comes default with Node.js".
```

**Review by rarkins — CHANGES_REQUESTED** [review]

File: docs/usage/node.md (hunk 5 above)
Comment (HonkingGoose) [reply docs/usage/node.md:74]: I think something went wrong here?

I think we do need the ending ``` characters or the code highlight block won't work?

Comment (HonkingGoose) [reply docs/usage/node.md:74]: I'm sorry but I really don't understand what your suggesting here.
I think the GitHub interface is displaying your comment wrong?

**Pull request merged**